### PR TITLE
🐛 (contribution) Fix SQL model to build a Type 2 SCD to handle NULL cursor_field values correctly

### PR DESCRIPTION
**Pull request by ChristopheDuong**
## What
Wrap-up contribution from https://github.com/airbytehq/airbyte/pull/4802

## Recommended reading order
1. `airbyte-integrations/bases/base-normalization/normalization/transform_catalog/stream_processor.py`
2. the rest

## Pre-merge Checklist
Expand the checklist which is relevant for this PR. 

<details><summary> <strong> Connector checklist </strong> </summary>
<p>

- [x] Issue acceptance criteria met
- [x] PR name follows [PR naming conventions](https://docs.airbyte.io/contributing-to-airbyte/updating-documentation#issues-and-pull-requests)
- [x] Secrets are annotated with `airbyte_secret` in the connector's spec
- [x] Credentials added to Github CI if needed and not already present. [instructions for injecting secrets into CI](https://docs.airbyte.io/contributing-to-airbyte/building-new-connector#using-credentials-in-ci). 
- [x] Unit & integration tests added as appropriate (and are passing)
    * Community members: please provide proof of this succeeding locally e.g: screenshot or copy-paste acceptance test output. To run acceptance tests for a Python connector, follow instructions in the README. For java connectors run `./gradlew :airbyte-integrations:connectors:<name>:integrationTest`.
- [ ] `/test connector=connectors/<name>` command as documented [here](https://docs.airbyte.io/contributing-to-airbyte/building-new-connector#updating-an-existing-connector) is passing. 
    * Community members can skip this, Airbyters will run this for you. 
- [ ] Code reviews completed
- [ ] Documentation updated 
    - [ ] `README.md`
    - [ ] `docs/SUMMARY.md` if it's a new connector
    - [ ] Created or updated reference docs in `docs/integrations/<source or destination>/<name>`.
    - [ ] Changelog in the appropriate page in `docs/integrations/...`. See changelog [example](https://docs.airbyte.io/integrations/sources/stripe#changelog)
    - [ ] `docs/integrations/README.md` contains a reference to the new connector
    - [ ] Build status added to [build page](https://github.com/airbytehq/airbyte/blob/master/airbyte-integrations/builds.md)
- [ ] Build is successful
- [ ] Connector version bumped like described [here](https://docs.airbyte.io/contributing-to-airbyte/building-new-connector#updating-a-connector)
- [ ] New Connector version released on Dockerhub by running the `/publish` command described [here](https://docs.airbyte.io/contributing-to-airbyte/building-new-connector#updating-a-connector)
- [ ] No major blockers
- [ ] PR merged into master branch
- [ ] Follow up tickets have been created
- [ ] Associated tickets have been closed & stakeholders notified
</p>
</details>

### CODE REVIEW

File: airbyte-integrations/bases/base-normalization/integration_tests/dbt_integration_test.py [hunk 1]
```diff
@@ -160,8 +159,14 @@ def generate_profile_yaml_file(self, destination_type: DestinationType, test_roo
...
-            profiles_config["credentials_json"] = json.dumps(profiles_config)
-            profiles_config["dataset_id"] = self.target_schema
+            credentials = profiles_config
+            profiles_config = {
+                "credentials_json": json.dumps(credentials),
+                "dataset_id": self.target_schema,
+                "project_id": credentials["project_id"],
+            }
+        elif destination_type.value == DestinationType.MYSQL.value:
+            profiles_config["database"] = self.target_schema
```
Comment (ChristopheDuong) [inline airbyte-integrations/bases/base-normalization/integration_tests/dbt_integration_test.py:169]: These changes are necessary to fix tests due to: https://github.com/airbytehq/airbyte/pull/4699

File: airbyte-integrations/bases/base-java/build.gradle [hunk 2]
```diff
@@ -7,7 +7,6 @@ dependencies {
     implementation 'commons-cli:commons-cli:1.4'
 
     implementation project(':airbyte-protocol:models')
-    implementation project(':airbyte-queue')
```
Comment (ChristopheDuong) [inline airbyte-integrations/bases/base-java/build.gradle:10]: Removing this from https://github.com/airbytehq/airbyte/pull/4699/files#r672182180

**Comment by ChristopheDuong** [comment]
Why in the github, the ordering of gradle tasks differ from running on local?

It seems if I am not wrong that the test is being run before docker images are being built?
1. Running at `2021-07-21T13:52:32.9882761Z`
Screenshot 2021-07-21 at 16 24 23
2. Building at `2021-07-21T13:52:40.7024761Z`
Screenshot 2021-07-21 at 16 24 47

File: docs/understanding-airbyte/basic-normalization.md [hunk 3]
```diff
@@ -281,3 +281,12 @@ Note that all the choices made by Normalization as described in this documentati
 * to build a [custom SQL view](../operator-guides/transformation-and-normalization/transformations-with-sql.md) with your own naming conventions
 * to export, edit and run [custom dbt normalization](../operator-guides/transformation-and-normalization/transformations-with-dbt.md) yourself
 * or further, you can configure the use of a custom dbt project within Airbyte by following [this guide](../operator-guides/transformation-and-normalization/transformations-with-airbyte.md).
+
+## CHANGELOG
```
Comment (ChristopheDuong) [inline docs/understanding-airbyte/basic-normalization.md:285]: Is it good @avaidyanatha if I'm adding this similar to connectors?

Comment (avaidyanatha) [reply docs/understanding-airbyte/basic-normalization.md:285]: @ChristopheDuong Absolutely, this is good!

**Review by davinchia — APPROVED** [review]

**Pull request merged**